phpunit/class-block-templates-test.php

```diff
@@ -55,6 +78,25 @@ function test_gutenberg_build_template_result_from_file() {
 		$this->assertEquals( 'Single', $template->title );
 		$this->assertEquals( 'Used when a single entry that is not a Page is queried', $template->description );
 		$this->assertEquals( 'wp_template', $template->type );
+
+		// Test template parts.
+		$template_part = _gutenberg_build_template_result_from_file(
+			array(
+				'slug' => 'header',
+				'path' => __DIR__ . '/fixtures/template.html',
+			),
+			'wp_template_part'
+		);
+		$this->assertEquals( get_stylesheet() . '//header', $template_part->id );
+		$this->assertEquals( get_stylesheet(), $template_part->theme );
+		$this->assertEquals( 'header', $template_part->slug );
+		$this->assertEquals( 'publish', $template_part->status );
+		$this->assertEquals( false, $template_part->is_custom );
+		$this->assertEquals( 'header', $template_part->title );
+		$this->assertEquals( '', $template_part->description );
+		$this->assertEquals( 'wp_template_part', $template_part->type );
+		// TODO - update null to 'header' once tt1-blocks theme.json updated for template part section info.
```

Comment:
Feels weird that we are depending on the TT1B theme for unit tests. Isn't there a way to create a theme fixture and use that for unit testing? (Not a blocker, but it would be a great follow up)

Reply:
> Isn't there a way to create a theme fixture and use that for unit testing?

🤔  Im not sure the best way to go about that.  It seems like it would need to be installed and activate-able like a regular theme, with the same file structures.  And at that point maybe using TT1B just makes the most sense (as opposed to creating and maintaining a separate one just for testing) since it is the most kept up-to-date block-based theme that we are developing FSE around currently.  I wonder if @youknowriad has given this any thought previously?

Reply:
It makes more sense for E2E tests. I don't think it's reasonable for unit tests. We opened an issue to discuss it: https://github.com/WordPress/gutenberg/issues/28692 Would love to hear your thoughts. It's not a blocker, just a thought 😄